lms/models/organization.py

```diff
@@ -0,0 +1,80 @@
+from base64 import urlsafe_b64encode
+from uuid import uuid4
+
+import sqlalchemy as sa
+from sqlalchemy.dialects.postgresql import JSONB
+
+from lms.db import BASE
+from lms.models._mixins import CreatedUpdatedMixin
+from lms.models.application_settings import ApplicationSettings
+from lms.models.region import Region
+
+
+class Organization(CreatedUpdatedMixin, BASE):
+    """Model for Organizations comprised of application instances."""
+
+    __tablename__ = "organization"
+
+    id = sa.Column(sa.Integer(), autoincrement=True, primary_key=True)
+
+    _vanity_id = sa.Column(
+        sa.UnicodeText(),
+        nullable=False,
+        unique=True,
+        # We don't use a standard UUID-4 format here as they are common in Tool
+        # Consumer Instance GUIDs, and might be confused for them. These also
+        # happen to be shorter and guaranteed URL safe.
+        default=lambda: urlsafe_b64encode(uuid4().bytes).decode("ascii").rstrip("="),
+    )
+    """
+    A human readable URL safe vanity id.
+
+    The vanity id can be customised by the user and therefore can end-up not
+    being _globally_ unique, only locally unique to the LMS instance. For this
+    reason `public_id` should be used instead.
+    """
+
+    def public_id(self, region: Region) -> str:
+        """
+        Get the globally unique id which also indicates the region.
+
+        This returns vales with prefixes like 'us:' or 'ca:'.
+
+        This is useful if you only have the id, but don't know which region you
+        should be looking for it in and is the only id suitable for sharing
+        outside a single region LMS context.
+        """
+
+        # We use '.' as the separator here because it's not in base64, but it
+        # is URL safe. The other option is '~'.
+        # See: https://www.ietf.org/rfc/rfc3986.txt (2.3)
+        # >    unreserved  = ALPHA / DIGIT / "-" / "." / "_" / "~"
+        return region.code + "." + self._vanity_id
```

Comment:
We might want to use the same thing for other entities.

- Should we move this to `_columns.py`? I reckon not until we have the second usage.

- Should we include the "what" in the public ID? `what:where:id`

`org:us:AAAAAAAA`

Then we could use the exact same setup in other objects, get a:

`user:us:BBBBBBB`

I'm not thinking about collision here, just about making them more "human friendly". It won't make difference in most cases but it might be useful when seeing IDs in the wild and difficult to do after we ship the first of these public_ids